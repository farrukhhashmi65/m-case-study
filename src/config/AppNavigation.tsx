import React from 'react';
import { NavigationContainer, DefaultTheme } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import RegisterUser from '../features/registerUser';
import Dashboard from '../features/dashboard';
import * as Routes from './routes';
import { ThemeProvider } from "styled-components";
import {
  SafeAreaView,
  StyleSheet
} from 'react-native';
import { useSelector } from 'react-redux'

const Stack = createNativeStackNavigator();
const AppNavigation = () => {

  const {
    theme
  } = useSelector((state: any) => state.themeReducer)
  const styles = getStyles(theme)

  // App Navigation
  const MyTheme = {
    ...DefaultTheme,
    colors: {
      ...DefaultTheme.colors,
      background: theme.PRIMARY_BACKGROUND_COLOR
    },
  };

  return (
    <ThemeProvider theme={theme}>
      <SafeAreaView style={styles.safeAreaContainer}>
        <NavigationContainer theme={MyTheme} >
          <Stack.Navigator
            screenOptions={{
              headerShown: false,
            }}
          >
            <Stack.Screen
              name={Routes.ROUTE_USER_REGISTRATION}
              component={RegisterUser}
            />
            <Stack.Screen
              name={Routes.ROUTE_USER_DASHBOARD}
              component={Dashboard}
            />
          </Stack.Navigator>
        </NavigationContainer>
      </SafeAreaView>
    </ThemeProvider>
  );
};

const getStyles = (theme : any) => StyleSheet.create({
  safeAreaContainer: {
      flex: 1,
      backgroundColor: theme.PRIMARY_BACKGROUND_COLOR
  },
})

export default AppNavigation;