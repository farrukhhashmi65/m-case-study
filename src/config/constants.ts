
export const Languages = ["en", 'ru', 'fr']

export enum Countries {
    UAE = "UAE",
    India = "India",
    Pakistan = "Pakistan",
    Oman = "Oman"
}

export const CountryImages: any = {
    India: require('../assets/India.png'),
    Pakistan: require('../assets/Pakistan.png'),
    Oman: require('../assets/Oman.png'),
    UAE: require('../assets/UAE.png')
}

export enum APIStatus {
    Error = "error",
    Success = "success",
}