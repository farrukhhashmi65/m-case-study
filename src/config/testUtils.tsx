import React, { PropsWithChildren } from 'react';
import { ThemeProvider } from 'styled-components/native';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux'
import { setupStore } from '../redux/store'
import { render } from '@testing-library/react-native';


export const mockTheme = {
  // Your theme object goes here
  PRIMARY_TEXT_COLOR: '#3d3d3d',
  ERROR_TEXT_COLOR: '#cc0000',
  PRIMARY_TEXT_COLOR_LIGHT: '#797979',
  PRIMARY_COLOR_FAINT: '#FFF3E0',
  PRIMARY_COLOR_LIGHT: '#FFB74D',
  DISBALED_COLOR: '#D3D3D3',
  PRIMARY_COLOR: "#FF9800",
  SECONDARY_TEXT_COLOR: "#ffffff"
};


export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: any = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
          <NavigationContainer>
            <ThemeProvider theme={mockTheme}>
              {children}
            </ThemeProvider>
          </NavigationContainer>
      </Provider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}



