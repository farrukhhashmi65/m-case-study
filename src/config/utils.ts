
import i18n from 'i18next'
import { I18nManager } from 'react-native';
import RNRestart from 'react-native-restart'; 
import { base, themePalettes } from "../config/theme";
import { Countries } from './constants';

export const setLanguage = async (languageKey: string) => {
    const isNewLanguageRTL = i18n.dir(languageKey) === 'rtl';
    
    const isCurrentLayoutRTL = I18nManager.isRTL;
    
    const isLayoutChangeNeeded = isCurrentLayoutRTL !== isNewLanguageRTL;
   
    i18n.changeLanguage(languageKey)
    if (isLayoutChangeNeeded) {    
        I18nManager.forceRTL(isNewLanguageRTL); // successful, since otherwise, the new
        RNRestart.restart();                    // language would not apply after restart.
    }
};

export const getTheme = (country?: string) => {
  switch (country) {
    case Countries.UAE:
      return { ...base, ...themePalettes.orange }
    case Countries.India:
      return { ...base, ...themePalettes.blue }
    case Countries.Pakistan:
      return { ...base, ...themePalettes.green }
    case Countries.Oman:
      return { ...base, ...themePalettes.cyan }
    default:
      return { ...base, ...themePalettes.teal };
  }
}
