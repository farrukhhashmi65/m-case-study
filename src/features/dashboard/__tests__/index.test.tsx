import React from 'react';
import Dashboard from '../index';
import { renderWithProviders, mockTheme } from '../../../config/testUtils';

jest.mock('styled-components', () => ({
    ...jest.requireActual('styled-components'),
    useTheme: () => (mockTheme),
}));


jest.mock('react-i18next', () => ({
    // mock useTranslation hook 
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: jest.fn(),
                language: 'EN'
            }
        }
    }
}))

test('renders dashboard', () => {
    const { toJSON  } = renderWithProviders(<Dashboard/>);
  expect(toJSON()).toMatchSnapshot();

});