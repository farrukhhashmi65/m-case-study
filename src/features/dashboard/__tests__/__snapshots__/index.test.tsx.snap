// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dashboard 1`] = `
<View
  style={
    {
      "flex": 1,
      "paddingHorizontal": 22,
    }
  }
>
  <Text
    color="#3d3d3d"
    style={
      [
        {
          "color": "#3d3d3d",
        },
        {
          "fontSize": 18,
          "fontWeight": "600",
        },
      ]
    }
  >
    welcome
  </Text>
</View>
`;
